src/counts.jl

```diff
@@ -9,53 +9,81 @@
 typealias IntUnitRange{T<:Integer} UnitRange{T}
 
 #### functions for counting a single list of integers (1D)
-
+"""
+    addcounts!(r, x, levels[, wv])
+
+Fill the array `r` with the number of occurrences in `x` of
+each value in `levels`. If a weighting vector `wv` is specified,
+the counts are incremented by the corresponding weights rather
+than by 1.
+"""
 function addcounts!(r::AbstractArray, x::IntegerArray, levels::IntUnitRange)
-	# add counts of integers from x to r
-
-	k = length(levels)
-	length(r) == k || raise_dimerror()
-
-	m0 = levels[1]
-	m1 = levels[end]
-	b = m0 - 1
-
-	@inbounds for i in 1 : length(x)
-		xi = x[i]
-		if m0 <= xi <= m1
-			r[xi - b] += 1
-		end
-	end
-	return r
+    # add counts of integers from x to r
+
+    k = length(levels)
+    length(r) == k || throw(DimensionMismatch())
+
+    m0 = levels[1]
+    m1 = levels[end]
+    b = m0 - 1
+
+    @inbounds for i in 1 : length(x)
+        xi = x[i]
+        if m0 <= xi <= m1
+            r[xi - b] += 1
+        end
+    end
+    return r
 end
 
 function addcounts!(r::AbstractArray, x::IntegerArray, levels::IntUnitRange, wv::WeightVec)
-	k = length(levels)
-	length(r) == k || raise_dimerror()
-
-	m0 = levels[1]
-	m1 = levels[end]
-	b = m0 - 1
-	w = values(wv)
-
-	@inbounds for i in 1 : length(x)
-		xi = x[i]
-		if m0 <= xi <= m1
-			r[xi - b] += w[i]
-		end
-	end
-	return r
+    k = length(levels)
+    length(r) == k || throw(DimensionMismatch())
+
+    m0 = levels[1]
+    m1 = levels[end]
+    b = m0 - 1
+    w = values(wv)
+
+    @inbounds for i in 1 : length(x)
+        xi = x[i]
+        if m0 <= xi <= m1
+            r[xi - b] += w[i]
+        end
+    end
+    return r
 end
 
-counts(x::IntegerArray, levels::IntUnitRange) = addcounts!(zeros(Int, length(levels)), x, levels)
-counts(x::IntegerArray, levels::IntUnitRange, wv::WeightVec) = addcounts!(zeros(eltype(wv), length(levels)), x, levels, wv)
+
+"""
+    counts(x, [levels,] [k,] [wv]) -> Vector
```

Comment:
Oh, that's a thing now? I thought that was still under discussion. Isn't the issue/PR still open on Julia?